### ISSUE TITLE
Mixed air node temp does not match heat recovery outlet temp in 100% OA/DOAS systems

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4592]
Helpdesk ticket 9836
User noted that the HX outlet temp which feeds directly into the OA mixer was not the same as the mixed air temp for a 100% OA system.  The mass flows check out, but in the user file, the mixed air node temp is lagging the HX outlet temp by one system timestep.  Tested this with HVACTemplate-5ZoneFanCoil-DOAS.idf example file and found that sometimes it matches, sometimes it lags and sometimes it's neither.  

Searching back, If found https://github.com/NREL/EnergyPlus/issues/2105 where there were problems with mass flow rates with multiple HX and the fix by @wfbuhl  was:

[Jan 15 2008] WFB

For outside air systems with heat exchangers I've implemented an additional pass back
 up the list of components. This gets rid of the mass flow imbalnace warnings. Still getting
 humidity ratio warnings, but I don't think these have anything to do with this CR.

I have not checked to see if this is what current code does, but that could explain why the HX might update its outlet temp after the OA mixer has been simulated.  May need to go up and back down the list?

**Comment by rraustad** [comment]
The temperatures seems to match when there is flow?

![hxtemps](https://cloud.githubusercontent.com/assets/5778156/11448715/3fd4205a-952c-11e5-9c51-ca9f7fb33a5c.png)

**Comment by rraustad** [comment]
OK, the temps do lag if you look close.

![hxtemps_simu_sys_1](https://cloud.githubusercontent.com/assets/5778156/11448800/ba9aa64a-952e-11e5-8f26-9b272efb689f.png)

**Comment by rraustad** [comment]
The original CR was to align the temps and mass flow rate when a HX was used in the OA system.
Since the OA equipment is simulated from OA node to mixed node, the HX is simulated before the mixer and the temp/flow at the HX when it is simulated is from the last iteration. So a second loop was added to run back up the OA equipment list from mixed node to OA node.

I added a 3rd loop to traverse the OA equipment list again to see if that helps.

```
for ( CompNum = 1; CompNum <= OutsideAirSys( OASysNum ).NumComponents; ++CompNum ) {
    CompType = OutsideAirSys( OASysNum ).ComponentType( CompNum );
    CompName = OutsideAirSys( OASysNum ).ComponentName( CompNum );
    SimOAComponent( CompType, CompName, OutsideAirSys( OASysNum ).ComponentType_Num( CompNum ), FirstHVACIteration, OutsideAirSys( OASysNum ).ComponentIndex( CompNum ), AirLoopNum, Sim, OASysNum, OAHeatCoil, OACoolCoil, OAHX );
    if ( OAHX ) ReSim = true;
}
// if there were heat exchangers and/or desiccant wheel in the OA path, need to simulate again
// in reverse order to propagate the air flow and conditions out the relief air path to the relief air
// exit node
if ( ReSim ) {
    for ( CompNum = OutsideAirSys( OASysNum ).NumComponents - 1; CompNum >= 1; --CompNum ) {
        CompType = OutsideAirSys( OASysNum ).ComponentType( CompNum );
        CompName = OutsideAirSys( OASysNum ).ComponentName( CompNum );
        SimOAComponent( CompType, CompName, OutsideAirSys( OASysNum ).ComponentType_Num( CompNum ), FirstHVACIteration, OutsideAirSys( OASysNum ).ComponentIndex( CompNum ), AirLoopNum, Sim, OASysNum, OAHeatCoil, OACoolCoil, OAHX );
    }
    for( CompNum = 1; CompNum <= OutsideAirSys( OASysNum ).NumComponents; ++CompNum ) {
        CompType = OutsideAirSys( OASysNum ).ComponentType( CompNum );
        CompName = OutsideAirSys( OASysNum ).ComponentName( CompNum );
        SimOAComponent( CompType, CompName, OutsideAirSys( OASysNum ).ComponentType_Num( CompNum ), FirstHVACIteration, OutsideAirSys( OASysNum ).ComponentIndex( CompNum ), AirLoopNum, Sim, OASysNum, OAHeatCoil, OACoolCoil, OAHX );
    }
}
```

The results show the temps match now.

![hxtemps_simu_sys_1_fixed](https://cloud.githubusercontent.com/assets/5778156/11448908/126ce8c6-9532-11e5-9219-cee4b414ccaf.png)

**Comment by rraustad** [comment]
I added a 2nd HX in the OA system expecting the temps to be misaligned. They were not so this seems to be a reasonable fix. I was thinking that the mixer OA node and exhaust node need to be tested to ensure the OA system component performance has converged, maybe that would be overkill (i.e., loop until temps converged).

```
AirLoopHVAC:OutdoorAirSystem:EquipmentList,
  OA Sys 1 Equipment list, !- Name
  HeatExchanger:AirToAir:FlatPlate,  !- Component 1 Object Type
  OA Heat Exchanger 2,     !- Component 1 Name
  HeatExchanger:AirToAir:FlatPlate,  !- Component 1 Object Type
  OA Heat Exchanger 1,     !- Component 1 Name
  OutdoorAir:Mixer,        !- Component 2 Object Type
  OA Mixer;                !- Component 2 Name
```

![hxtemps_simu_sys_1_2hxs_fixed](https://cloud.githubusercontent.com/assets/5778156/11449025/46fe8d08-9535-11e5-8134-288278b4108d.png)

**Comment by rraustad** [comment]
PR #5337

**Comment by rraustad** [comment]
Very small differences so showing comparison of temperature difference between V8.4 and this fix.

![mixertempdiffs](https://cloud.githubusercontent.com/assets/5778156/11452727/20b73e98-95c0-11e5-8454-366e6cd73f3f.png)

**Comment by mjwitte** [comment]
Closed via #5337
